pkg/action/failback.go

```diff
@@ -0,0 +1,213 @@
+		if present {
+			for _, migration := range policyMigration {
+				if !isMigrationComplete(migration.Status) {
+					msg := fmt.Sprintf("Waiting for the completion of migration %s", migration.Name)
+					log.ActionLog(action).Infof(msg)
+					ac.recorder.Event(action,
+						v1.EventTypeWarning,
+						string(storkv1.ActionStatusScheduled),
+						msg)
+					ac.updateAction(action)
+					return
+				}
+			}
+		}
+	}
+
+	// check if it has any latest successful migration
+	schedulePolicyType, latestMigration := getLatestMigrationPolicyAndStatus(*migrationSchedule)
+	if latestMigration.Status == storkv1.MigrationStatusFailed {
+		msg := fmt.Sprintf("The latest migration %s is in %s state, hence aborting failback", latestMigration.Name, storkv1.MigrationStatusFailed)
+		log.ActionLog(action).Errorf(msg)
+		action.Status.Status = storkv1.ActionStatusFailed
+		action.Status.Reason = msg
+		ac.recorder.Event(action,
+			v1.EventTypeWarning,
+			string(storkv1.ActionStatusFailed),
+			msg)
+		ac.updateAction(action)
+		return
+	}
+
+	if latestMigration.Status == storkv1.MigrationStatusSuccessful ||
+		latestMigration.Status == storkv1.MigrationStatusPartialSuccess {
+		// Check if the latest migration has run with in the policy time
+		alreadyTriggered, err := schedule.AlreadyTriggered(
+			migrationSchedule.Spec.SchedulePolicyName,
+			migrationSchedule.Namespace,
+			schedulePolicyType,
+			latestMigration.FinishTimestamp,
+		)
+		if err != nil {
+			msg := fmt.Sprintf("Error while checking if the latest migration %s has been successful: %v", latestMigration.Name, err)
+			log.ActionLog(action).Errorf(msg)
+			action.Status.Status = storkv1.ActionStatusFailed
+			action.Status.Reason = msg
+			ac.recorder.Event(action,
+				v1.EventTypeWarning,
+				string(storkv1.ActionStatusFailed),
+				msg)
+			ac.updateAction(action)
+			return
+		}
+		if alreadyTriggered {
+			log.ActionLog(action).Infof("The latest migration %s is in threshold range", latestMigration.Name)
+			// Suspend migration schedule
+			suspend := true
+			if *migrationSchedule.Spec.Suspend {
+				migrationSchedule.Spec.Suspend = &suspend
+			}
+			msg := fmt.Sprintf("Suspending migration schedule %s before doing failback operation", migrationSchedule.Name)
+			log.ActionLog(action).Infof(msg)
+			_, err := storkops.Instance().UpdateMigrationSchedule(migrationSchedule)
+			if err != nil {
+				log.ActionLog(action).Errorf("Error suspending migration schedule %s: %v", migrationSchedule.Name, err)
+			}
+			action.Status.Status = storkv1.ActionStatusSuccessful
+			ac.updateAction(action)
+		} else {
+			// Fail the action failback if latest successful migration is older than latestMigrationThresholdTime
+			// And migrationschedule is in suspended state
+			log.ActionLog(action).Infof("The latest migration %s is not in threshold range", latestMigration.Name)
+			msg := fmt.Sprintf("Failing failback operation as the latest migration %s is not in threshold range", latestMigration.Name)
```

Comment:
nit..threshold time range. 
This is just to be explicit and avoid confusion. Also are we going to publish the latestMigrationThresholdTime in our docs?
Maybe we can just say threshold time range of 1 hr...

Reply:
Now it is not 1 hr time. It's based on the schedule policy. We are looking for a successful migration within the time range ofallowed earliest time based on schedule policy from now and now.

I have modified the message.